src/util/invariant.h

```diff
@@ -242,6 +252,10 @@ inline void invariant_violated_string(
 // changed by a previous method call.
 // "The contract of the previous method call means the following
 //  condition holds".
+//
+// A difference between CHECK_RETURN and POSTCONDITION is that CHECK_RETURN is
+// supposed to assert much more specific details about what a function has
+// returned, that is dependent on the specific needs of a caller.
```

Comment:
The difference between `POST_CONDITION` and `CHECK_RETURN` isn't the details, it's that one is a statement of a function about itself, whereas the other is a statement about what a caller expects from a function.